File: src/main/ts/proxy/SearchEngine.ts
interface SearchEngine {
    search(query: string): string;
}

class GoogleSearchEngine implements SearchEngine {
    search(query: string): string {
        console.log(`Performing Google search for query: ${query}`);
        return `Google search results for query: ${query}`;
    }
}

var searchEngine: GoogleSearchEngine = new GoogleSearchEngine();
var result = searchEngine.search("NEWS");

File: java/src/main/java/com/epam/eng/pattern/proxy/SearchEngine.java
package com.epam.eng.pattern.proxy;

public interface SearchEngine {
    String search(String query);
}

class GoogleSearchEngine implements SearchEngine {
    public String search(String query) {
        System.out.println("Performing Google search for query: " + query);
        return "Google search results for query: " + query;
    }
}

// todo: consider to create a Search Engine Proxy class, which is responsible to call GoogleSearchEngine